packages/hub/tasks/notify-customer-payment.ts

```diff
@@ -0,0 +1,34 @@
+import { inject } from '@cardstack/di';
+import CardpaySDKService from '../services/cardpay-sdk';
+import Web3HttpService from '../services/web3-http';
+import WorkerClient from '../services/worker-client';
+
+interface CustomerPaymentSubscriptionEvent {
+  returnValues: {
+    merchantSafe: string;
+    spendAmount: string;
+  };
+}
+
+export default class NotifyCustomerPayment {
+  cardpay: CardpaySDKService = inject('cardpay');
+  web3: Web3HttpService = inject('web3-http', { as: 'web3' });
+  workerClient: WorkerClient = inject('worker-client', { as: 'workerClient' });
+
+  async perform(payload: CustomerPaymentSubscriptionEvent) {
+    let Safes = await this.cardpay.getSDK('Safes', this.web3.getInstance());
+    let { safe } = await Safes.viewSafe(payload.returnValues.merchantSafe);
```

Comment:
I think it's a lot easier to just query the subgraph for the payment event, that will include both the merchant, customer and amount paid in both the issuing token (useful for the merchant) and in SPEND.

Reply:
Here is an example of a query that gets teh EOA for the prepaid card, the EOA of the merchant, issuing token amount, issuing token name, and SPEND amount: ![Screen Shot 2021-12-01 at 6 22 37 PM](https://user-images.githubusercontent.com/61075/144330100-833bdbae-600b-43d5-afcb-3130a87bf5e0.png)